sabnzbd/sorting.py

```diff
@@ -797,218 +546,180 @@ def filter_files(_file, current_path):
-        mapping.append(("%_title", self.date_info["title_three"]))
-
-        mapping.append(("%t", self.date_info["title"]))
-        mapping.append(("%.t", self.date_info["title_two"]))
-        mapping.append(("%_t", self.date_info["title_three"]))
-
-        mapping.append(("%sn", self.date_info["ttitle"]))
-        mapping.append(("%s.n", self.date_info["ttitle_two"]))
-        mapping.append(("%s_n", self.date_info["ttitle_three"]))
-        mapping.append(("%sN", self.date_info["title"]))
-        mapping.append(("%s.N", self.date_info["title_two"]))
-        mapping.append(("%s_N", self.date_info["title_three"]))
-
-        # Replace year
-        mapping.append(("%year", self.date_info["year"]))
-        mapping.append(("%y", self.date_info["year"]))
-
-        # Replace resolution
-        mapping.append(("%r", self.date_info["resolution"]))
-
-        if self.date_info["ep_name"]:
-            mapping.append(("%desc", self.date_info["ep_name"]))
-            mapping.append(("%.desc", self.date_info["ep_name_two"]))
-            mapping.append(("%_desc", self.date_info["ep_name_three"]))
+def guess_what(name: str, sort_type: Optional[str] = None) -> MatchesDict:
+    """Guess metadata for movies or episodes from their name. The sort_type ('movie' or 'episode')
+    is passed as a hint to guessit, if given."""
+
+    if not name:
+        raise ValueError("Need a name for guessing")
+
+    # Remove any passwords from the name
+    name = scan_password(name)[0]
+
+    # Avoid trouble with names starting with a digit (esp. with no year in the name)
+    digit_fix = "FIX" if name[0].isdigit() else ""
+
+    guessit_options = {
+        # "no-user-config": True,
+        "expected_title": [],  # This isn't empty by default?
+        # "allowed_countries": [],
+        # "allowed_languages": [],
+        "excludes": EXCLUDED_GUESSIT_PROPERTIES,
+    }
+    if sort_type:
+        # Hint the type if known
+        guessit_options["type"] = sort_type
+
+    guess = guessit.api.guessit(digit_fix + name, options=guessit_options)
+    logging.debug("Initial guess for %s is %s", digit_fix + name, guess)
+
+    if digit_fix:
+        # Unfix the title
+        guess["title"] = guess.get("title", "")[len(digit_fix) :]
+
+    # Force season to 1 for seasonless episodes with no date
+    if guess.get("type") == "episode" and not "date" in guess:
+        guess.setdefault("season", 1)
+
+    # Try to avoid setting the type to movie on arbitrary jobs (e.g. 'Setup.exe') just because guessit defaults to that
+    table = str.maketrans({char: "" for char in whitespace + "_.-()[]{}"})
+    if guess.get("type") == "movie" and not sort_type == "movie":  # No movie hint
+        if (
+            guess.get("title", "").translate(table) == name.translate(table)  # Check for full name used as title
+            or any(
+                c in guess.get("release_group", "") for c in (whitespace + punctuation)
```

Comment:
I don't get this part? Why would that make it `unkown`?
In general (also due to the black-formatting) this `if` statement is a bit hard to read.

Reply:
Because it's an indication the guessing is (way) off, see the earlier example of guessit using the release group as a dumping ground. Unless there's groupnames that include whitespace or punctuation?

The formatting is entirely black's doing: the original had one condition per line, with comments directly adjacent.